daemon/src/monitor.rs

```diff
@@ -525,3 +529,275 @@ where
+                cfd_actor_addr: address,
+                client: stub::Client::default(),
+                latest_block_height: BlockHeight(0),
+                current_status: BTreeMap::default(),
+                awaiting_status: HashMap::from_iter(subscriptions),
+            }
+        }
+    }
+
+    fn txid1() -> Txid {
+        "1278ef8104c2f63c03d4d52bace29bed28bd5e664e67543735ddc95a39bfdc0f"
+            .parse()
+            .unwrap()
+    }
+
+    fn script1() -> Script {
+        "6a4c50001d97ca0002d3829148f63cc8ee21241e3f1c5eaee58781dd45a7d814710fac571b92aadff583e85d5a295f61856f469b401efe615657bf040c32f1000065bce011a420ca9ea3657fff154d95d1a95c".parse().unwrap()
+    }
+
+    #[derive(Default)]
+    struct MessageRecordingActor {
+        events: Vec<Event>,
+    }
+
+    impl xtra::Actor for MessageRecordingActor {}
+
+    #[async_trait]
+    impl xtra::Handler<Event> for MessageRecordingActor {
+        async fn handle(&mut self, message: Event, _ctx: &mut xtra::Context<Self>) {
+            self.events.push(message);
+        }
+    }
+
+    mod stub {
+        use super::*;
+        use bdk::electrum_client::ScriptStatus;
+
+        #[derive(Default)]
+        pub struct Client {
+            transactions: HashMap<Txid, i32>,
+            block_height: usize,
+        }
+
+        impl Client {
+            pub fn include_tx(&mut self, tx: Txid, height: i32) {
+                self.transactions.insert(tx, height);
+            }
+
+            pub fn advance_to_height(&mut self, height: usize) {
+                self.block_height = height;
+            }
+        }
+
+        impl ElectrumApi for Client {
+            fn block_headers_subscribe(&self) -> Result<HeaderNotification, Error> {
+                Ok(HeaderNotification {
+                    height: self.block_height,
+                    header: block::BlockHeader {
+                        version: 0,
+                        prev_blockhash: Default::default(),
+                        merkle_root: Default::default(),
+                        time: 0,
+                        bits: 0,
+                        nonce: 0,
+                    },
+                })
+            }
+
+            fn batch_script_get_history<'s, I>(
+                &self,
+                _: I,
+            ) -> Result<Vec<Vec<GetHistoryRes>>, Error>
+            where
+                I: IntoIterator<Item = &'s Script> + Clone,
+            {
+                Ok(self
+                    .transactions
+                    .iter()
+                    .map(|(tx, included_at)| {
+                        vec![GetHistoryRes {
+                            height: *included_at,
+                            tx_hash: *tx,
+                            fee: None,
+                        }]
+                    })
+                    .collect())
+            }
+
+            fn batch_call(&self, _batch: &Batch) -> Result<Vec<serde_json::Value>, Error> {
+                todo!()
```

Comment:
nit: maybe use `unreachable` instead? I always feel `todo` indicates something that should be actioned at some point

Reply:
This was just autofilled by rust-analyzer but yeah, I can change it to unreachable!